Sources/PackageGraph/CMakeLists.txt

```diff
@@ -15,6 +15,9 @@ add_library(PackageGraph
   PinsStore.swift
   Pubgrub.swift
   RawPackageConstraints.swift
+  ResolvedPackage.swift
+  ResolvedProduct.swift
+  ResolvedTarget.swift
```

Comment:
I'm not sure I agree with moving these out of PackageModel since it makes sense to keep them close to their non-resolved counterparts.

Reply:
Despite their correspondence in name, I think there's a stronger case to be made for moving them into PackageGraph.

Functionally, the resolved models aren't referenced at all in the PackageModel module aside from their declaration. Instead, they're called exclusively by way of `PackageGraphLoader.load` and the delegated `PackageGraph` initializer.

Conceptually, the concept of a "resolved" package / product / target goes beyond modeling a package. You can only resolve something within the context of a package graph.

Reply:
My rationale is more around the fact that since these types wrap parts of their non-resolved counterparts, they might need to get updated in tandem. Having them in the same module could help suggest that, but maybe we should have a shared protocol instead that both types implement. That way we could "force" updates to the resolved types.

Reply:
Indeed, code that encapsulates other code requires coordination when the underlying assumptions change. But this is only really a problem when you don't control what's on both sides of the API boundary. In practice, if you're changing one part of SPM, you're usually making those changes in tandem with other parts of SPM that consume those APIs.

If SPM modules were separate, interdependent projects, then you'd likely want to use an abstraction like the protocol you suggested. But here, we have something closer to a monorepo, where the boundaries are more conceptual than practical, so you can usually get away without too much indirection.

All of that said, I found this change to be helpful in delineating responsibilities across modules (which is complicated by PackageDescription and PackageModel being so similar). But if that's in any way a blocker, I'd sooner revert that in the interest of keeping things moving forward — just let me know.

Reply:
I think you have convinced me now that it makes sense to move these types.